src/ORM/Exception/MissingAssociationException.php

```diff
@@ -0,0 +1,24 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @since         3.0.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\ORM\Exception;
+
+use Cake\Core\Exception\Exception;
+
+/**
+ * Exception raised when marshaller hits nonexistend association while marshalling entity
+ */
+class MissingAssociationException extends Exception
```

Comment:
Do we need a discrete type for this? When contain fails we use `InvalidArgumentException`. Any reason we couldn't do that in this scenario as well?